### PR TITLE
Setup scraper

### PR DESCRIPTION
**Pull request by SakshumGusain**
## Issues solved
- Scraped all commodities, state and district names and stored in a JSON

### CODE REVIEW

**Comment by SakshumGusain** [comment]
- Created script for getting data for single day
- Created API database for data storage and setup Github CI for automatically storing data daily at 8 AM

File: .github/workflows/download-spec.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+name: Updating Database
+
+on:
+  push:
```
Comment (ChakshuGautam) [inline .github/workflows/download-spec.yml]: Add a schedule rather than push. Every day.

File: 06-02-2024-data.csv
Comment (ChakshuGautam) [inline 06-02-2024-data.csv:1]: Add to .gitignore

File: 07-02-2024-data.csv
Comment (ChakshuGautam) [inline 07-02-2024-data.csv:1]: Add to .gitignore

File: 08-02-2024-data.csv
Comment (ChakshuGautam) [inline 08-02-2024-data.csv:1]: Add to .gitignore

File: commoditiesAPI.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+                  data.records !== null &&
+                  data.records != undefined
+                ) {
+                  console.log(data.records)
+                  createTable(data)
+                  insertRows(db, data);
+                }
+              })
+              .catch((error) => console.error("Error:", error));
+      }, j*1000)
```
Comment (ChakshuGautam) [inline commoditiesAPI.js:64]: Increase timeout. It should not look like we are trying to DDoS it.

File: commoditiesAPI.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    }
+  }, delay*1000)
```
Comment (ChakshuGautam) [inline commoditiesAPI.js:66]: Increase this as well.

File: commoditiesAPI.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+
+// Creating a variable for accessing database
+const db = new sqlite3.Database("./API.db", sqlite3.OPEN_READWRITE, (err) => {
+  if (err) console.log(err);
+});
+
+const apiKey = process.env.COMMODITIES_API_KEY;
+const apiUrl =
+  "https://api.data.gov.in/resource/9ef84268-d588-465a-a308-a864a43d0070";
```
Comment (ChakshuGautam) [inline commoditiesAPI.js:17]: Extract this to a config.

File: commodities.json
Comment (ChakshuGautam) [inline commodities.json:1]: Move this to a `masters` folder.

File: states-and-districts.json
Comment (ChakshuGautam) [inline states-and-districts.json:1]: Move this to a `masters` folder.

File: index.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    for (let i = 1; i < rowColumns.length; i++) {
+        const name = $(rowColumns.eq(i)).text().trim();
+        if (name) {
+            commodities.push(name);
+        }
+    }
+
+  });
+
+  console.log(commodities)
```
Comment (ChakshuGautam) [inline index.js:50]: Remove console logs - add winston.

File: index.js [hunk 6]
```diff
@@ -0,0 +1,65 @@
+
+import cheerio from "cheerio";
+import axios from "axios";
+
+async function performScraping() {
+  const result = await axios.request({
+    method: "GET",
+    url: "https://agmarknet.gov.in/OtherPages/CommodityList.aspx",
```
Comment (ChakshuGautam) [inline index.js:8]: This can do down. Let's add a failure/retry here.

**Review by ChakshuGautam — COMMENTED** [review]
Add a readme. Mostly cover what is already there in the workflow.

File: .github/workflows/download-spec.yml
Comment (ChakshuGautam) [inline .github/workflows/download-spec.yml:1]: Change file name to `scape.yml1

**Comment by SakshumGusain** [comment]
- Resolved all the issues mentioned

**Pull request merged**